web/src/click_handlers.ts

```diff
@@ -983,6 +983,33 @@ export function initialize(): void {
         }
     });
 
+    const compose_buttons = tippy.default(".compose_button_tooltip", {
+        trigger: "mouseenter",
+        placement: "top",
+    });
+
+    tippy.createSingleton(compose_buttons, {
+        onTrigger(instance, event) {
+            const currentTarget = event.currentTarget;
+            if (currentTarget instanceof HTMLElement) {
+                const templateId = currentTarget.dataset.tooltipTemplateId;
+                if (templateId) {
+                    const templateElement = document.querySelector(`#${templateId}`);
+                    if (templateElement) {
+                        const parsedHTML = ui_util.parse_html(templateElement.innerHTML);
+                        instance.setContent(parsedHTML);
+                    }
+                }
+
+                if (currentTarget.classList?.contains("disabled-on-hover")) {
+                    instance.setProps({delay: 0});
+                } else {
+                    instance.setProps({delay: 500});
+                }
+            }
+        },
```

Comment:
We also need to handle the cleanup for the singletons. Here we would need to destroy the children tippy instances after removing the parent singleton instance.

According to the [docs](https://atomiks.github.io/tippyjs/v6/addons/#destroy):

> When you call singleton.destroy(), the tippyInstances you passed as an argument will not be destroyed also. They are separate instances that can be reused again elsewhere. You should also destroy the tippy instances upon cleanup.